hack/build

```diff
@@ -6,12 +6,16 @@ set -e
 : "${PACKAGE=github.com/docker/buildx}"
 : "${VERSION=$(./hack/git-meta version)}"
 : "${REVISION=$(./hack/git-meta revision)}"
+: "${PRERELEASE=$(./hack/git-meta version | rev | cut -c3- | rev | cut -d'-' -f 2 | grep -E '(^tp|^rc).*')}"
+: "${REVISION_NUMBER=1}"
+VERSION=$(printf "v%sm%s%s" "$(./hack/git-meta version | rev | cut -c3- | rev | cut -d'-' -f 1 | cut -c2-)" "${REVISION_NUMBER}" "-${PRERELEASE}")
```

Comment:
I wasn't sure where the mversion was going to be included from going forward, I can include that from the git tag instead.

Reply:
The _package revision_ is going to be provided from the packaging pipeline going forward. I don't think it makes sense for the packaging pipeline to pass the [software version](https://mirantis.jira.com/wiki/spaces/ENGINE/pages/5037555737/MCR+versioning#Software-versions) into the build script piecemeal. @aepifanov's [release process proposal](https://docs.google.com/document/d/1_qTtaIpLdJW2Hy5jlktssSGo2nOKcd6mAYHzTnrxdN4/edit) -- which you reviewed with the rest of the team -- is for the packaging pipeline to take a Git tag name as a parameter and use the tag name as the version of the software being built. Since the tag is going to encode the full software version; revision, prerelease and all; there would be little value in passing the version components separately. The pipeline would have to parse the version string, only to have the build script re-concatenate it into an identical string.

Note that we will be creating multiple tags which reference the same commit, e.g. to rebuild an RC as a GA. That means we can't rely on the `hack/git-meta version` command. It looks for which tags reference the commit, but if more than one tag references the commit it might choose the wrong tag. This `hack/build` script already affords overriding the automatically inferred software version by setting the `VERSION` environment variable. Why not simply leverage that, and have our packaging script set `VERSION` to the git tag name when invoking the build?